source/plugin/permissions.rst

```diff
@@ -228,14 +228,65 @@ mechanisms themselves.
 Contexts
 ========
 
-If you consider each permission to a privilege or ability to be able to do something, a Context is the circumstances
+If you consider each permission to a privilege or ability to be able to do something, a ``Context`` is the circumstances
 where that privilege is usable.
 
-You might want to give a Subject permission to do something, but only when the Subject is in a certain world, or in a
-certain Region.
+You might want to give a ``Subject`` permission to do something, but only when the ``Subject`` is in a certain world,
+or in a certain region.
 
-Contexts are accumulated by a Subject, and are then used by the PermissionService to decide if the Subject has a
-privilege or not.
+Contexts are accumulated by a ``Subject``, and are then used by the ``PermissionService`` to decide if the ``Subject``
+has a privilege or not.
 
 Sponge provides some contexts by default, but it is generally down to other plugins to provide additional contexts to
-the PermissionService, through a ContextCalculator.
\ No newline at end of file
+the PermissionService, through a ``ContextCalculator``.
+
+When creating contexts for your own plugin please try to avoid conflicts with other plugins (ex by choosing a unique
+context key) unless these contexts are meant to be shared.
+
+.. note::
+    
+    Please make sure that your ``ContextCalculator`` responds as fast as possible as it will get called frequently.
+
+Example
+~~~~~~~
+
+Your ``ContextCalculator`` may look like this:
+
+.. code-block:: java
+
+    public class MyArenaPlugin implements ContextCalculator<Subject> {
+    
+        private final Context IN_ANY_ARENA = new Context("myArenaPlugin", "*");
+    
+        @Override
+        public void accumulateContexts(Subject subject, Set<Context> accumulator) {
```

Comment:
Not a great example IMO. A better example would be to map the name of the arena to the Context, instead of a *, and then only add the context to the accumulator if the user was in an arena.

They're meant to be used as key value pairs, not just a collection of keys.

So:

```
if (user in arena):
  add context where key="myArenaPlugin_arena", value="arenaName"
```